### ISSUE TITLE
[Feature request] Allow override of default playback speed on a per-channel basis

### ISSUE THREAD

**Issue by zpz5HAU-tgc3fgw2xwr** [omarroth/invidious#3093]
**Is your feature request related to a problem? Please describe.**
The way I currently use Invidious, my playback speed defaults to 1.25x. This works well for most videos, but there are some channels that I watch daily that I always set 1x playback speed on. It would be nice to have that override saved on a per-channel basis.

**Describe the solution you'd like**
For any channels you subscribe to, you have an option to override the default playback speed. Or maybe just save the last used playback speed for any channels you're subscribed to.

**Describe alternatives you've considered**
The only alternative is to remember to manually change the playback speed.

**Comment by SamantazFox** [comment]
Imo, that feature would be too complex to implement. Preferences are already very complex, implementing per-channel preferences would clutter the backend (code and database storage).

**Comment by ngInit** [comment]
We're already storing channels we're subscribed to in the database, it should be pretty simple to add another option to that existing list. Since something like that probably doesn't already exist, it can also be implemented in a way that can easily be expanded upon for future features.

And anybody self hosting their own instance wouldn't care about a minimal increase in DB storage... so if your issue with storage is for the public instances, then we can add a flag to enable this as a optional feature in the environment INVIDIOUS_CONFIG.

**Comment by SamantazFox** [comment]
> We're already storing channels we've subscribed to in the database, it should be pretty simple to add another option to that existing list. Since something like that probably doesn't already exist, it can also be implemented in a way that can easily be expanded upon for future features.
> 
> And anybody self hosting their own instance wouldn't care about a minimal increase in DB storage... so if your issue with storage is for the public instances, then we can add a flag to enable this as a optional feature in the environment INVIDIOUS_CONFIG.

That's that easy to do. Not without a huge performance impact on large instances, at least. That's easy to do with a single user subscribed to 5-10 channels, that's another deal when you have 20 000 users and at least as many subscribed channels...

**Comment by ngInit** [comment]
Right. So then it can be a config option like I mentioned. Most people running this don't have 20,000 users.

**Comment by ngInit** [comment]
If database changes are the issue here, another option could be just saving settings like this, and possibly others in the future, as an optional cookie. Not ideal if you clear your browser cache, but still a move in the right direction.

I'm not trying to be confrontational about this, but being told this is too difficult/complex doesn't make any sense. I'd much rather hear that this isn't a developer priority rather than "this is too hard". Because it doesn't have to be hard.

**Comment by ngInit** [comment]
Here, as a proof of concept and to show I'm not talking out my ass I did the work for you in about 20 minutes using TamperMonkey. It's not as difficult as you're making it out to be. Feel free to use and modify this however you need to.

In the meantime, I suppose I'll stick to modifying my script. But I'm sure this functionality would prove useful to others if it was implemented officially.

```
// @require      http://code.jquery.com/jquery-3.4.1.min.js
// ==/UserScript==
const $ = window.jQuery;

const customSettings = {
    channels: {
        "Philip DeFranco": {
            playback_speed: "1x",
        },
    },
};

$(document).ready(function() {
    "use strict";
    Object.keys(customSettings.channels).forEach(key => {
        if ($("#channel-name")[0].innerText.includes(key)) {
            const setting = customSettings.channels[key];
            const playbackSpeedOptions = $(".vjs-playback-rate ul > li");
            for (let i = 0; i < playbackSpeedOptions.length; i++) {
                const speedOption = playbackSpeedOptions[i];
                speedOption.innerText.includes(setting.playback_speed) ? speedOption.click() : 0;
            }
        }
    });
});
```
